### PR TITLE
Add description field to tags triggers and variables

### LINKED ISSUE #249: Add description/comment field to Tags/Triggers/Variables for better documentation of functionality
This feature would come in handy when more than one person is administering the TagManager.

### PR DESCRIPTION
**Pull request by snake14**
### Description:

Added description field for variables, tags, and triggers.
Fixes: #249 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: Updates/4.12.0-b2.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+     * migrations directly in `doUpdate()`, they won't be displayed to the user. Migrations will be executed in the
+     * order as positioned in the returned array.
+     *
+     * @param Updater $updater
+     * @return Migration\Db[]
+     */
+    public function getMigrations(Updater $updater)
+    {
+        // many different migrations are available to be used via $this->migration factory
+        $migration1 = $this->migration->db->changeColumnType('log_visit', 'example', 'BOOLEAN NOT NULL');
```
Comment (tsteur) [inline Updates/4.12.0-b2.php]: fyi these 2 migrations can be removed

**Review by tsteur — APPROVED** [review]
Nice, well done @snake14 👍  this worked for me. I've also double checked that the generated container JS won't include any of the entered description information and it doesn't. 

Looks all good from my perspective.

Note: For testing need to increase `core/Version.php` to version `b2` so that the tables are being updated.

@AltamashShaikh did you also want to have a quick look? Otherwise looks good to merge for me 🎉

File: Updates/4.12.0-b2.php (hunk 1 above)
Comment (snake14) [reply Updates/4.12.0-b2.php]: @tsteur Thank you for catching that. I just removed them.

**Comment by AltamashShaikh** [comment]
@tsteur Changes are fine, was just wondering should we keep the description field as a `textarea` instead of a `input` box ?

**Comment by AltamashShaikh** [comment]
@snake14 Also can you merge 4.x-dev branch in this branch and just keep your changes for `umd.js` and `umd.min.js` file and later just rebuild the vue files and it will regenerate a fresh version of both the files

**Comment by tsteur** [comment]
> Changes are fine, was just wondering should we keep the description field as a textarea instead of a input box ?

Textarea may be better indeed 👍

**Comment by snake14** [comment]
> @tsteur Changes are fine, was just wondering should we keep the description field as a `textarea` instead of a `input` box ?

@AltamashShaikh @tsteur  Thank you. A textarea makes sense. I was modelling my changes after the description field in VersionEdit.vue, which uses an input, but I can use textarea for my new fields. Should I update the description input in the Version edit form to a textarea as well?

**Comment by AltamashShaikh** [comment]
> was modelling my changes after the des

@snake14 yes, update it to textarea

**Comment by snake14** [comment]
@AltamashShaikh While I was updating the screenshots, I found a bug. I'm guessing that it was introduced in the Vue conversion and the screenshot difference was small enough that it didn't get caught previously. The Environment select wasn't defaulting to another value when pushing to live wasn't allowed. I went ahead and fixed that. Please let me know if the fix looks good.

File: vue/src/Version/VersionEdit.vue [hunk 2]
```diff
@@ -312,8 +312,10 @@ export default defineComponent({
 
         if (this.canPublishToLive) {
           this.version.environments = ['live'];
-        } else if (!this.version.environments) {
```
Comment (AltamashShaikh) [inline vue/src/Version/VersionEdit.vue:315]: @snake14 Why did we remove this check, not able to understand

**Review by AltamashShaikh — COMMENTED** [review]
@snake14 All looks good :+1:  and works as expected.
Just left 1 comment for my understanding

File: vue/src/Version/VersionEdit.vue (hunk 2 above)
Comment (snake14) [reply vue/src/Version/VersionEdit.vue:315]: @AltamashShaikh I noticed that `this.version` is defined on line 306 and doesn't include the `environments` property, so `this.version.environments` is always going to be undefined and doesn't need to be checked.

Comment (AltamashShaikh) [reply vue/src/Version/VersionEdit.vue:315]: understood :+1:

**Review by AltamashShaikh — APPROVED** [review]

**Pull request merged**